lampod/src/ln/peer_manager.rs

```diff
@@ -151,18 +152,36 @@ impl LampoPeerManager {
             async_run!(async move {
                 let bind_addr = format!("0.0.0.0:{}", listen_port);
                 log::info!(target: "lampo", "Litening for in-bound connection on {bind_addr}");
-                let listener = tokio::net::TcpListener::bind(bind_addr).await.unwrap();
+                let listener = match tokio::net::TcpListener::bind(bind_addr).await {
+                    Ok(listener) => listener,
+                    Err(e) => {
+                        log::error!(target: "lampo", "Error binding to address: {}", e);
+                        return Err::<(), error::Error>(e.into());
+                    }
+                };
+
                 loop {
                     let peer_manager = peer_manager.clone();
-                    let tcp_stream = listener.accept().await.unwrap().0;
-                    log::info!(target: "lampo", "Got new connection {}", tcp_stream.peer_addr().unwrap());
-                    let _ = tokio::spawn(async move {
-                        // Use LDK's supplied networking battery to facilitate inbound
-                        // connections.
-                        net::setup_inbound(peer_manager.clone(), tcp_stream.into_std().unwrap())
+
+                    match listener.accept().await {
```

Comment:
```suggestion
                    match listener.accept().await.map_err(|err| error::anyhow!("Error accepting connection: {err}"))? {
```

I add the previous suggestion by changing your code, but I think you can just use this `map_or` syntax.

In addition, this syntax is also used a lot in the code